### DISCUSSION TITLE
How to build PIVPY locally?

### DISCUSSION THREAD

**Discussion by nepomnyi** [alexlib/pivpy#58] [Q&A] [answered]
Dr. Liberzon @alexlib , I am wondering if you can be so kind to share the way you build PIVPY locally, please.
You see, when I took software development course, I was taught how to build packages locally using a setup.py file. But PIVPY doesn't have a setup.py file, and I don't know how to build a package without a setup.py file. Internet search gave me some results, but not something certain.
So, imagine your are in the PIVPY development conda environment. What is the sequence of commands you would type into the console to build PIVPY locally?
Thank you in advance.
Ivan

**Comment by alexlib** [comment, marked as answer]
Hi @nepomnyi 

I use: 
1) `conda` for environments
2) `pip` for installations

so:

`conda create -n pivpy` 
`conda activate pivpy`
`conda install pip`
`pip install -e .` 

the `pip install -e .` is the local installation for development. `-e` ensures that if I change the code and run `pivpy` in the same environment I do not need to reinstall every time. 

at the end I package it and upload to `pypi.org` as a binary wheel

**Reply by nepomnyi** [reply]
Thank you very much! I will try it and see how it goes.
Yesterday, I, also, was having troubles with building OpenPIV as well (even though OpenPIV has a setup.py file). I received a `distutils` error. So, I guess, I will try your approach both with OpenPIV and PIVPY and I hope it will work.
Ivan
